extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/MicrometerCountedInterceptor.java

```diff
@@ -59,16 +63,30 @@ Object countedMethod(ArcInvocationContext context) throws Exception {
         Method method = context.getMethod();
         Tags commonTags = getCommonTags(method.getDeclaringClass().getName(), method.getName());
 
-        // If we're working with a CompletionStage
-        final boolean stopWhenCompleted = CompletionStage.class.isAssignableFrom(method.getReturnType());
-        if (stopWhenCompleted) {
+        Class<?> returnType = method.getReturnType();
+        if (TypesUtil.isCompletionStage(returnType)) {
             try {
-                return ((CompletionStage<?>) context.proceed()).whenComplete((result, throwable) -> {
-                    recordCompletionResult(counted, commonTags, throwable);
+                return ((CompletionStage<?>) context.proceed()).whenComplete(new BiConsumer<Object, Throwable>() {
```

Comment:
The problem with this approach is that you don't catch the cancellation, which means you would not record if the operation is canceled. I believe it's fine if it's documented (like do not use completion stage/ completable future, ever).

Reply:
Hm... I didn't actually change this part, I only removed the lambda. So I'll let @ebullient fix it